packages/pages/src/common/src/function/internal/getFunctionFilepaths.ts

```diff
@@ -4,27 +4,43 @@ import { FunctionFilePath } from "./types.js";
 
 /**
  * Get all the functions files in a directory
- * @param root the directory to check for functions
+ * @param root the directory to check for functions (src/functions or dist/functions)
  * @returns a list of function filepaths
  */
 export const getFunctionFilepaths = (root: string): FunctionFilePath[] => {
-  const functionsRoot = new Path(root);
+  const functionsRoot = new Path(root); // resolve the functions root
+  // Get all js/ts files in the directory or subdirectories
   const filepaths = glob.sync(
     `${functionsRoot.getAbsolutePath()}/**/*.{tsx,jsx,js,ts}`
   );
 
   return filepaths.map((filepath) => {
+    // This RegExp captures anything after the functions root and before the file extension
     const filePathMatcher = filepath.match(`${root}(.*)\\.`);
     if (!filePathMatcher || filePathMatcher.length < 2) {
+      // the matcher should be able to find a relative path
+      // with [0] being the absolute path and [1] being the relative path
       throw new Error(
         `Cannot resolve relative path from ${root} for ${filepath}`
       );
     }
+
     return {
+      // absolute: /Users/example/Desktop/mySite/src/functions/http/getInfo.ts
       absolute: filepath,
```

Comment:
Thinking about this, it would be easier to just use  path.parse(filepath) to get some of the other things. Relative might be the only one you need to calculate yourself still. It would be much clearer though.

Reply:
Additionally, maybe we should just return the path.ParsedPath too instead of each one of these things.